src/adc.rs

```diff
@@ -535,6 +537,11 @@ impl Channels {
     {
         self.flags |= 0x1 << C::channel()
     }
+
+    /// constructor for empty channels object
+    pub fn new() -> Self {
```

Comment:
I think it would be nicer to have have the constructor come first in the `impl` block, but that's not important.